Makefile

```diff
@@ -130,8 +130,9 @@ verify-modules: get-deps
 		echo "-- Verifying modules for $$i --"; \
 		working_dir=`pwd`; \
 		cd $${i}; \
+		go mod tidy; \
```

Comment:
We should probably update get-deps to call `go mod tidy` instead of having the mix of those calling `go mod download` and/or `go mod tidy` in that step, then this step again calling tidy.

Reply:
That could be a follow up though, since this is already fixing the issues found with the go.sum files and has reviews from some of the various teams.

Reply:
I think having to call `go mod tidy` twice here is the symptom of `go mod tidy` not downloading packages if you don't already have them in your go-bin. Might be worth experimenting with this on a fresh vm to see how it behaves. Really, it should be 

1. Call `go mod download` to get any new packages that the user may not have
2. Call `goo mod tidy` to clean up since `download` may inevitably create transient dependencies

I think this experience also got better with go 1.17 since they optimized some of how go mod download works (and hopefully won't download / track a bunch of transient dependencies)

Reply:
Created a follow up PR #2514 . Resolving this conversation